hydra-node/src/Hydra/API/Server.hs

```diff
@@ -98,20 +98,20 @@ withAPIServer config env party persistence tracer chain pparams serverOutputFilt
     commitInfoP <- mkProjection CannotCommit projectCommitInfo
     headIdP <- mkProjection Nothing projectInitializingHeadId
     pendingDepositsP <- mkProjection [] projectPendingDeposits
+    history <- newTVarIO []
     _ <-
       runConduitRes $
         source
-          .| mapC output
-          .| iterM (lift . atomically . update headStatusP)
-          .| iterM (lift . atomically . update snapshotUtxoP)
-          .| iterM (lift . atomically . update commitInfoP)
-          .| iterM (lift . atomically . update headIdP)
-          .| mapM_C (lift . atomically . update pendingDepositsP)
-
-    -- FIXME: this is not streaming, it loads all events into memory
```

Comment:
@v0d1ch This is not fixing this issue. The result is still a list of full length in `history`.

Also, instead of `mapM_C` and `modifyTVar` you could just use `sinkList` and make `history` the result of the whole conduit pipeline.

Reply:
yes, the history TVar still holds all of the data but the memory usage in Server should be smaller if we don't materialize the whole list and then put it in a TVar, don't you think so?